syntax = "proto3";

package user;

service UserService {
  rpc CreateUser (CreateUserRequest) returns (User);
  rpc GetUser (GetUserRequest) returns (User);
  rpc UpdateUser (UpdateUserRequest) returns (User);
  rpc DeleteUser (DeleteUserRequest) returns (DeleteUserResponse);
  rpc GetUsers (GetUsersRequest) returns (GetUsersResponse);
}

message User {
  int32 id = 1;
  string name = 2;
  string email = 3;
  string phone = 4;
  string created_at = 5;
  string updated_at = 6;
}

message CreateUserRequest {
  string name = 1;
  string email = 2;
  string phone = 3;
}

message GetUserRequest {
  int32 id = 1;
}

message UpdateUserRequest {
  int32 id = 1;
  string name = 2;
  string email = 3;
  string phone = 4;
}

message DeleteUserRequest {
  int32 id = 1;
}

message DeleteUserResponse {
  bool success = 1;
  string message = 2;
}

message GetUsersRequest {
  int32 page = 1;
  int32 limit = 2;
}

message GetUsersResponse {
  repeated User users = 1;
  int32 total = 2;
}